src/services/api.ts

```diff
@@ -47,3 +145,92 @@ export async function getServerPublicKey(): Promise<string> {
 
 	return await response.data.ocs.data['public-key']
 }
+
+/**
+ * Set a folder as encrypted.
+ *
+ * @param folderId - The fileid of the folder
+ */
+export async function setFolderAsEncrypted(folderId: number): Promise<void> {
+	const url = generateOcsUrl(Url.Encrypted, { folderId })
+	await axios.put(
+		url,
+		{},
+		{ headers: { 'X-E2EE-SUPPORTED': 'true' } },
+	)
+}
+
+/**
+ * Lock a folder for metadata updates.
+ *
+ * @param folderId - The folder to lock
+ * @param counter - The incremented counter value (see metadata)
+ * @param shareToken - Optional share token if folder is a share (to identify the owner)
+ * @param token - Optional existing token to extend the lock
+ * @return The locking token
+ */
+export async function lockFolder(folderId: number, counter: number, token?: string, shareToken?: string): Promise<string> {
+	const url = generateOcsUrl(Url.Lock, { folderId })
+	const response = await axios.post<OCSResponse<{ 'e2e-token': string }>>(
+		url,
+		{
+			'e2e-token': token,
+			shareToken,
+		},
+		{
+			headers: {
+				'X-NC-E2EE-COUNTER': counter.toString(),
+				'X-E2EE-SUPPORTED': 'true',
+			},
+		},
+	)
+	return response.data.ocs.data['e2e-token']
+}
+
+/**
+ * Unlock a folder for metadata updates.
+ *
+ * @param folderId - The folder to lock
+ * @param token - Optional existing token to extend the lock
+ * @param shareToken - Optional share token if folder is a share (to identify the owner)
+ * @param abort - Whether to abort the ongoing operation
+ */
+export async function unlockFolder(folderId: number, token: string, shareToken?: string, abort?: true): Promise<void> {
+	const url = generateOcsUrl(Url.Lock, { folderId })
+	await axios.delete<OCSResponse<{ 'e2e-token': string }>>(
+		url,
+		{
+			data: {
+				abort: abort ? 'true' : undefined,
+				shareToken,
+			},
+			headers: {
+				'E2E-TOKEN': token,
+				'X-E2EE-SUPPORTED': 'true',
+			},
+		},
+	)
+}
+
+/**
+ * Set initial metadata for a folder.
+ *
+ * @param fileId - The fileid of the folder
+ * @param metaData - The metadata to set
+ * @param token - The locking token
+ * @param signature - The signature to verify the request
+ */
+export async function createMetadata(fileId: number, metaData: string, token: string, signature: string): Promise<void> {
+	const url = generateOcsUrl(Url.Metadata, { fileId })
+	await axios.post(
+		url,
+		{ metaData, token },
```

Comment:
I assumed it is linked with your comment below?
```suggestion
		{ metaData },
```